common/src/main/java/org/wso2/testgrid/common/Utils.java

```diff
@@ -80,6 +81,59 @@ public static boolean executeCommand(String command, File workingDirectory) thro
         }
     }
 
+    /**
+     * Executes a command.
+     * Used to execute a script with given deployment details as environment variables.
+     *
+     * @param command          Command to execute.
+     * @param workingDirectory Directory the command is executed.
+     * @param deployment       Deployment details for environment.
+     * @return Returns the output of script execution as a String.
+     * @throws CommandExecutionException When there is an error executing the command.
+     */
+    public static String executeCommand(String command, File workingDirectory, Deployment deployment)
```

Comment:
Instead of keeping this method in the `Utils` class, shall we move it to a class called `ShellExecutor` since it serves a unique purpose? WDYT?

Reply:
+1
Made an issue #212